modyn/supervisor/internal/pipeline_executor/evaluation_executor.py

```diff
@@ -0,0 +1,331 @@
+        log.info = SingleEvaluationInfo(eval_request=eval_req)
+        request = GRPCHandler.prepare_evaluation_request(
+            dataset_config.model_dump(by_alias=True),
+            eval_req.model_id,
+            self.pipeline.evaluation.device,
+            eval_req.interval_start,
+            eval_req.interval_end,
+        )
+        response: EvaluateModelResponse = self.grpc.evaluator.evaluate_model(request)
+        if not response.evaluation_started:
+            log.info.failure_reason = EvaluationAbortedReason.DESCRIPTOR.values_by_number[
+                response.eval_aborted_reason
+            ].name
+            logger.error(
+                f"Evaluation for model {eval_req.model_id} on split {eval_req.interval_start} to "
+                f"{eval_req.interval_end} not started with reason: {log.info.failure_reason}."
+            )
+            return
+
+        logger.info(
+            f"Evaluation started for model {eval_req.model_id} on split {eval_req.interval_start} "
+            f"to {eval_req.interval_end}."
+        )
+        reporter = EvaluationStatusReporter(
+            eval_status_queue,
+            response.evaluation_id,
+            eval_req.dataset_id,
+            response.dataset_size,
+        )
+        evaluation = {response.evaluation_id: reporter}
+        reporter.create_tracker()
+        self.grpc.wait_for_evaluation_completion(eval_req.training_id, evaluation)
+
+        eval_result_writer = JsonResultWriter(self.pipeline_id, eval_req.trigger_id, self.pipeline_logdir)
+        self.grpc.store_evaluation_results([eval_result_writer], evaluation)
+        self.grpc.cleanup_evaluations([int(i) for i in evaluation])
+        assert isinstance(eval_result_writer, JsonResultWriter)
+
+        log.info.results = (
+            eval_result_writer.results["datasets"][0][dataset_config.dataset_id]
+            if eval_result_writer.results["datasets"]
+            else {}
+        )
+
+
+# ------------------------------------------------------------------------------------ #
+#                                       DevTools                                       #
+# ------------------------------------------------------------------------------------ #
+
+if __name__ == "__main__":
+    snapshot_path = Path(input("Enter eval snapshot path to (re)run evaluation executor: "))
+    if not snapshot_path.exists():
+        print("Path not found")
+        sys.exit(1)
+
+    # restart evaluation executor
+    ex = EvaluationExecutor.init_from_path(snapshot_path)
+
+    logs_ = PipelineLogs(
+        pipeline_id=ex.pipeline_id,
+        pipeline_stages={},
+        config=ConfigLogs(system=ex.config, pipeline=ex.pipeline),
+        experiment=True,
+        supervisor_logs=SupervisorLogs(),
+    )
+
+    logs_.supervisor_logs = ex.run_post_pipeline_evaluations(eval_status_queue=Queue())
+    logs_.materialize(snapshot_path, mode="final")
```

Comment:
will this overwrite existing log files? can we somehow merge that/extend the existing pipeline log file or is that a big effort?

Reply:
So far, I just did that manually. I didn't want this could destroy my main log file. It's enough to simply append the `stage_runs` section from one file to the other. Once this logic has proven itself I am happy to merge the two logfiles automatically.